src/content/docs/en/extend/add-content-collections.mdx

```diff
@@ -0,0 +1,326 @@
+    const { Content } = await entry.render();
+    ---
+    ```
+    
+
+8. Render your post `<Content />` within the layout for Markdown pages. This allows you to specify a common layout for all of your posts.
+
+    ```astro title="src/pages/posts/[...slug].astro" ins={15-17}
+    ---
+    import { getCollection } from 'astro:content';
+    import MarkdownPostLayout from '../../layouts/MarkdownPostLayout.astro';
+
+    export async function getStaticPaths() {
+      const blogEntries = await getCollection('posts');
+      return blogEntries.map(entry => ({
+        params: { slug: entry.slug }, props: { entry },
+      }));
+    }
+
+    const { entry } = Astro.props;
+    const { Content } = await entry.render();
+    ---
+    <MarkdownPostLayout frontmatter={entry.data}>
+      <Content />
+    </MarkdownPostLayout>
+    ```
+
+9. Remove the `layout` definition in each individual post's frontmatter. Your content is now wrapped in a layout when rendered, and this property is no longer needed.
+
+    ```md title="src/content/posts/post-1.md" del={2}
+    ---
+    layout: ../../layouts/MarkdownPostLayout.astro
+    title: 'My First Blog Post'
+    pubDate: 2022-07-01
+    ...
+    ---
+    ```
+
+### Replace `Astro.glob()` with `getCollection()`
+
+10. Anywhere you have a list of blog posts, like the tutorial's Blog page (`src/pages/blog.astro/`), you will need to replace `Astro.glob()` with [`getCollection()`](/en/reference/api-reference/#getcollection) as the way to fetch content and metadata from your Markdown files.
+
+      ```astro title="src/pages/blog.astro" "post.data" "getCollection(\"posts\")" "/posts/${post.slug}/" del={7} ins={2,8}
+    ---
+    import { getCollection } from "astro:content";
+    import BaseLayout from "../layouts/BaseLayout.astro";
+    import BlogPost from "../components/BlogPost.astro";
+
+    const pageTitle = "My Astro Learning Blog";
+    const allPosts = await Astro.glob("../pages/posts/*.md");
+    const allPosts = await getCollection("posts");
+    ---
+
+11. You will also need to update references to the data returned for each `post`. You will now find your frontmatter values on the `data` property of each object. Also, when using collections each `post` object will have a page `slug`, not a full URL.
+
+    ```astro title="src/pages/blog.astro" "data" "/posts/$\{post.slug\}/" del={14} ins={15}
+    ---
+    import { getCollection } from "astro:content";
+    import BaseLayout from "../layouts/BaseLayout.astro";
+    import BlogPost from "../components/BlogPost.astro";
+
+    const pageTitle = "My Astro Learning Blog";
+    const allPosts = await getCollection("posts");
+    ---
+    <BaseLayout pageTitle={pageTitle}>
+      <p>This is where I will post about my journey learning Astro.</p>
+      <ul>
+        {
+          allPosts.map((post) => (
+            <BlogPost url={post.url} title={post.frontmatter.title} />)}
+            <BlogPost url={`/posts/${post.slug}/`} title={post.data.title} />
```

Comment:
This is a great pattern @bogdaaamn ! I do like this! I think to change this, means we'd also have to change:

- the two code samples for this file (219 and 264)
- the syntax highlighting in both code samples (204 and 236)
- the text instructions in step 12 re: how to make a full page URL (232) (and maybe explain this property a tiny bit here?)
- the RSS link property (321)
Am I missing anything?  :sweat_smile: 

Since we do know the codebase they are starting from, I'm pretty confident that hard-coding in `/posts/` won't cause any problems here. But I really do like showing this pattern.

Reply:
We'll see where this lands for initial publish, because the hard-coded I think makes sense for the context coming from the very basic tutorial. This would also require us to change the final code in the blog-tutorial repo itself, because we do provide a final version to check against. So I might bookmark this and think of a separate PR to update at a later date.

Reply:
You're right! If I can contribute to either this or more TS content, would gladly do so! I will follow the news 👀